### PR TITLE
Improve appearance of Signup and User Profile fields

### PR DESCRIPTION
**Pull request by p2edwards**
Various WIP changes to `feature/tos` (#4699)

**Links**:
- [Internal thread for testing & feedback](https://chat.kobotoolbox.org/#narrow/stream/4-Kobo-Dev/topic/New.20required.20fields.20.28TOS.20update.29/near/308436)
- [Internal Notion card](https://www.notion.so/kobotoolbox/Require-extra-information-when-creating-accounts-epic-924597918852497da871e2d4de2181b3?p=7b3e54c54a8a48eb807241cf58e3e2f9&pm=s) (list of planned or considered changes)

### CODE REVIEW

File: jsapp/js/account/accountFieldsEditor.component.tsx [hunk 1]
```diff
@@ -183,14 +224,19 @@ export default function AccountFieldsEditor(props: AccountFieldsEditorProps) {
...
         )}
-      </div>
 
-      <div className={styles.row}>
+        {/* Insert a blank spacer to start a new row for these */}
+        {!!(count % 2) &&
+          isFieldToBeDisplayed('country') &&
+          isFieldToBeDisplayed('city') &&
+          ++count && <div className={styles.field} />}
+
```
Comment (LMNTL) [inline jsapp/js/account/accountFieldsEditor.component.tsx:233]: You might be able to accomplish the same thing that `count` and this empty `<div/>` are doing by sticking each row of 2 options in a flex container. The code works as-is, but some parts of it are hard to parse - for example, it took me a second to realize why it's `++count` instead of `count++`, and I was kind of surprised it was being incremented in the render.

File: jsapp/js/account/accountFieldsEditor.component.tsx [hunk 2]
```diff
@@ -366,15 +423,18 @@ export default function AccountFieldsEditor(props: AccountFieldsEditorProps) {
...
+              <label className={styles.checkboxLabel}>{t('Newsletter')}</label>
+              <Checkbox
+                checked={props.values.newsletter_subscription}
+                onChange={(isChecked: boolean) =>
+                  onAnyFieldChange('newsletter_subscription', isChecked)
+                }
+                label={getLabel('newsletter_subscription')}
+              />
+            </div>
+          </>
```
Comment (LMNTL) [inline jsapp/js/account/accountFieldsEditor.component.tsx:437]: The React fragment here `<></>` could probably get removed - there's only one child element.

**Review by LMNTL — CHANGES_REQUESTED** [review]

File: jsapp/js/account/accountFieldsEditor.component.tsx [hunk 3]
```diff
@@ -82,6 +91,33 @@ export default function AccountFieldsEditor(props: AccountFieldsEditorProps) {
...
+  const cleanedUrl = (value: string) => {
+    if (!value) {
+      return '';
+    }
+    value = ('' + value).trim();
+    if (!value.match(/.\../)) {
+      return value;
+    } // "dotless". don't change it
+    if (!value.match(/^https?:\/\/.*/)) {
+      value = 'https://' + value; // add missing protocol
```
[the remark this answers is not in the document]
Comment (p2edwards) [reply jsapp/js/account/accountFieldsEditor.component.tsx:103]: Agreed

File: jsapp/js/account/accountFieldsEditor.component.tsx (hunk 1 above)
Comment (LMNTL) [reply jsapp/js/account/accountFieldsEditor.component.tsx:235]: For this one, I'd be fine with some additional comments about the `count` logic and renaming it to something more descriptive (e.g. `fieldCount`.)

**Review by LMNTL — APPROVED** [review]
